inc/class-sitemaps-renderer.php

```diff
@@ -14,13 +14,13 @@ class Core_Sitemaps_Renderer {
 	 *
 	 * @param array $sitemaps List of sitemaps, see \Core_Sitemaps_Registry::$sitemaps.
 	 */
-	public function render_sitemapindex( $sitemaps ) {
+	public function render_index( $sitemaps ) {
 		header( 'Content-type: application/xml; charset=UTF-8' );
 		$sitemap_index = new SimpleXMLElement( '<?xml version="1.0" encoding="UTF-8" ?><sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"></sitemapindex>' );
 
 		foreach ( $sitemaps as $link ) {
 			$sitemap = $sitemap_index->addChild( 'sitemap' );
-			$sitemap->addChild( 'loc', esc_url( $link['slug'] ) );
+			$sitemap->addChild( 'loc', esc_url( $link->slug ) );
```

Comment:
<sitemapindex>
<sitemap>
<loc>http://posts</loc>
<lastmod>2004-10-01T18:23:17+00:00</lastmod>
</sitemap>
<sitemap>
<loc>http://page</loc>
<lastmod>2004-10-01T18:23:17+00:00</lastmod>
</sitemap>
</sitemapindex>

These links need to use get_sitemap_url()